lib/screens/customer_product_detail_page.dart

```diff
@@ -266,10 +315,11 @@ class _CustomerProductDetailPageState extends State<CustomerProductDetailPage> {
       child: GestureDetector(
         onTap: () {
           _addToCart();
-
-          _scaffoldKey.currentState.showSnackBar(
-            _buildAddedToCartNotification(),
-          );
+          if (_selectedItemColor != null && _selectedItemSize != null) {
```

Comment:
Please refer to my comment below. Snack bar should be shown whenever _addToCart succeeds:
```
if (_canAddToCart()) {
  _addToCart();
  _scaffoldKey.currentState.showSnackBar(      // <------- Add here!
    _buildAddedToCartNotification(),
  );
} else {
  setState(() {
    _showAddToCartFailedMessage = true;
  });
}
```